src/ServiceControl.Plugin.Nsb5.Heartbeat/Heartbeats.cs

```diff
@@ -12,96 +12,159 @@
     using NServiceBus.Unicast;
     using ServiceControl.Plugin.Heartbeat.Messages;
 
-    class Heartbeats : Feature, IWantToRunWhenConfigurationIsComplete
+    class Heartbeats : Feature
     {
-        public ISendMessages SendMessages { get; set; }
-        public Configure Configure { get; set; }
-
-        public UnicastBus UnicastBus { get; set; }
-
-        public CriticalError CriticalError { get; set; }
-
-        static ILog logger = LogManager.GetLogger(typeof(Heartbeats));
+        static ILog Logger = LogManager.GetLogger(typeof(Heartbeats));
         
         public Heartbeats()
         {
             EnableByDefault();
+
+            // we need a mechanism to start and stop to register and stop heartbeats timers.
+            // and both StartupTasks and IWantToRunWhenBusStartsAndStops aren't supported for send-only endpoints.
+            Prerequisite(context => !context.Settings.GetOrDefault<bool>("Endpoint.SendOnly"), "The Heartbeats plugin currently isn't supported for Send-Only endpoints");
+
+            RegisterStartupTask<HeartbeatStartup>();
         }
 
-        public void Run(Configure config)
+        protected override void Setup(FeatureConfigurationContext context)
         {
-            if (!IsEnabledByDefault)
+        }
+
+        class HeartbeatStartup : FeatureStartupTask, IDisposable
+        {
+            public HeartbeatStartup(ISendMessages sendMessages, Configure configure, UnicastBus unicastBus)
```

Comment:
tried that, didn't work out of the box. Does it require some special configuration?